solidity/src/Base7683.sol

```diff
@@ -3,82 +3,123 @@ pragma solidity 0.8.25;
     // ============ Libraries ============
     using SafeERC20 for IERC20;
 
     // ============ Constants ============
-
-    enum OrderStatus {
-        UNFILLED,
-        OPENED,
-        FILLED,
-        SETTLED,
-        REFUNDED
-    }
-
+    /// @notice The instance of the Permit2 contract.
     IPermit2 public immutable PERMIT2;
 
-    bytes32 public constant GASLESS_CROSS_CHAIN_ORDER_TYPEHASH = keccak256(
-        "GaslessCrossChainOrder(address originSettler,address user,uint256 nonce,uint64 originChainId,uint32 openDeadline,uint32 fillDeadline,bytes32 orderDataType,bytes orderData)"
+    /// @notice Type hash used for encoding ResolvedCrossChainOrder.
+    bytes32 public constant RESOLVED_CROSS_CHAIN_ORDER_TYPEHASH = keccak256(
+        "ResolvedCrossChainOrder(address user, uint64 originChainId, uint32 openDeadline, uint32 fillDeadline, Output[] maxSpent, Output[] minReceived, FillInstruction[] fillInstructions)Output(bytes32 token, uint256 amount, bytes32 recipient, uint64 chainId)FillInstruction(uint64 destinationChainId, bytes32 destinationSettler, bytes originData)"
     );
 
+    /// @notice The witness type string used in PERMIT2 transactions.
     string public constant witnessTypeString =
-        "GaslessCrossChainOrder witness)GaslessCrossChainOrder(address originSettler,address user,uint256 nonce,uint64 originChainId,uint32 openDeadline,uint32 fillDeadline,bytes32 orderDataType,bytes orderData)TokenPermissions(address token,uint256 amount)";
+        "ResolvedCrossChainOrder witness)ResolvedCrossChainOrder(address user, uint64 originChainId, uint32 openDeadline, uint32 fillDeadline, Output[] maxSpent, Output[] minReceived, FillInstruction[] fillInstructions)Output(bytes32 token, uint256 amount, bytes32 recipient, uint64 chainId)FillInstruction(uint64 destinationChainId, bytes32 destinationSettler, bytes originData)TokenPermissions(address token,uint256 amount)";
+
+    /// @notice Possible statuses for an order. Other possible statuses should be defined in the inheriting contract.
+    bytes32 public constant UNKNOWN = "";
+    bytes32 public constant OPENED = "OPENED";
+    bytes32 public constant FILLED = "FILLED";
+
+    // ============ Structs ============
+    /**
+     * @dev Represents data for an order that has been filled.
+     * @param originData The origin-specific data for the order.
+     * @param fillerData The filler-specific data for the order.
+     */
+    struct FilledOrder {
+        bytes originData;
+        bytes fillerData;
+    }
 
     // ============ Public Storage ============
 
-    mapping(address sender => uint256 nonce) public senderNonce;
+    /// @notice Tracks the bitmap of used nonces for each address.
+    mapping(address => mapping(uint256 => uint256)) public nonceBitmap;
 
-    mapping(bytes32 orderId => OrderData orderData) public orders;
+    /// @notice Stores the resolved orders by their ID.
+    mapping(bytes32 orderId => bytes resolvedOrder) public orders;
```

Comment:
This was already true previously, but curious why resolved orders have to be stored in storage?

Reply:
Maybe a good question for the original authors, but my understanding was that resolved orders can be passed at runtime to avoid writing, otherwise why do we have the concept of order resolution in the first place?

Reply:
My understanding was that the resolved order is for having a standard structure for the solvers to understand, so they don't have to deal with different types of `orderData` which are opaque.

We could save the orderData instead of the resolved order, and resolve again when handling the settle/refund, not sure what would be the more expensive though.

Reply:
yeah its a fair question, I'll ping them on a group